lib/AST/NameLookup.cpp

```diff
@@ -1633,29 +1634,52 @@ namelookup::lookupMacros(DeclContext *dc, DeclNameRef macroName,
   bool onlyMacros = static_cast<bool>(roles & getFreestandingMacroRoles()) &&
       !(roles - getFreestandingMacroRoles());
 
-  // Macro lookup should always exclude macro expansions; macro
-  // expansions cannot introduce new macro declarations. Note that
-  // the source location here doesn't matter.
-  UnqualifiedLookupDescriptor descriptor{
-    macroName, moduleScopeDC, SourceLoc(),
-    UnqualifiedLookupFlags::ExcludeMacroExpansions
-  };
-
-  if (onlyMacros)
-    descriptor.Options |= UnqualifiedLookupFlags::MacroLookup;
-
-  auto lookup = evaluateOrDefault(
-      ctx.evaluator, UnqualifiedLookupRequest{descriptor}, {});
-
-  for (const auto &found : lookup.allResults()) {
-    if (auto macro = dyn_cast<MacroDecl>(found.getValueDecl())) {
+  auto addChoiceIfApplicable = [&](ValueDecl *decl) {
+    if (auto macro = dyn_cast<MacroDecl>(decl)) {
       auto candidateRoles = macro->getMacroRoles();
       if ((candidateRoles && roles.contains(candidateRoles)) ||
           // FIXME: `externalMacro` should have all roles.
           macro->getBaseIdentifier().str() == "externalMacro") {
         choices.push_back(macro);
       }
     }
+  };
+
+  // When a module is specified, it's a module-qualified lookup.
+  if (moduleName) {
+    UnqualifiedLookupDescriptor moduleLookupDesc(
+        moduleName, moduleScopeDC, SourceLoc(),
+        UnqualifiedLookupFlags::TypeLookup);
+    auto moduleLookup = evaluateOrDefault(
+        ctx.evaluator, UnqualifiedLookupRequest{moduleLookupDesc}, {});
+    auto foundTypeDecl = moduleLookup.getSingleTypeResult();
+    auto *moduleDecl = dyn_cast_or_null<ModuleDecl>(foundTypeDecl);
+    if (!moduleDecl)
+      return {};
+
+    ModuleQualifiedLookupRequest req{moduleScopeDC, moduleDecl, macroName,
+                                     SourceLoc(), NL_ExcludeMacroExpansions};
```

Comment:
I think the lookup options should include `NL_OnlyMacros` when `onlyMacros` is `true`.

Reply:
I've added a few shadowing examples to `macro_shadowing.swift` to ensure that macros don't get shadowed by property wrappers. Therefore, I

1. added `ModuleLookup` to unqualified lookup options so that module name doesn't get shadowed when lookup qualified macro implementation
2. removed `onlyMacros` as attached macro should not be shadowed by local property wrappers, and always pass in `NL_OnlyMacros`/`UnqualifiedLookupFlags::MacroLookup` when looking up macros